test containers: add systemd to tang image

jbtrystram opened the pull request:
In the F41 fedora-minimal image, systemctl stopped working. We need to enable tang.service in the container so we need systemd in the container.

Adding systemd in the container.

dustymabe commented on tests/containers/tang/Containerfile:
```diff
@@ -1,6 +1,6 @@
 FROM registry.fedoraproject.org/fedora-minimal:41
```
would switching to the non-minimal image here be a better move? i.e. would less rpms get downloaded in the `dnf install` below?

jbtrystram replied:
it does not work

jbtrystram commented:
> > In the F41 fedora-minimal image, systemctl stopped working.
> 
> Do you mean that systemd was removed in the minimal images?

yeah sorry. But it was also removed from the non-minimal image so I just kept the minimal image and added systemd back
